src/timesync/timesyncd-bus.c

```diff
@@ -43,6 +45,60 @@ static int property_get_servers(
         return sd_bus_message_close_container(reply);
 }
 
+static int method_set_runtime_servers(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_strv_free_ char **valid_names = NULL;
+        _cleanup_strv_free_ char **msg_names = NULL;
+        Manager *m = userdata;
+        int r;
+
+        assert(m);
+        assert(message);
+
+        r = sd_bus_message_read_strv(message, &msg_names);
+        if (r < 0)
+                return r;
+
+        while (m->runtime_servers)
+                server_name_free(m->runtime_servers);
```

Comment:
you flush out the old servers before even just validating the new ones. Ideally we'd prepare the new servers fully before flushing old the old ones to make sure things either succeed or fail entirely but never leave half-completed stuff around if they fail midway. Now, I don't think we should be super strict on this, because it sometimes complicates code more than would be good. But still, we should at the very least try to do validation of user supplied settings before we make any changes.

here it's even quite easy, just exchange this loop with the next...

Reply:
Swapped the snippets now.